server/src/main/java/org/opensearch/action/support/replication/TransportWriteAction.java

```diff
@@ -82,6 +83,11 @@
 
     private final Function<IndexShard, String> executorFunction;
 
+    public static final Function<IndexShard, Boolean> IS_REMOTE_TXLOG_ENABLED = shard -> {
+        IndexSettings indexSettings = shard.indexSettings();
+        return indexSettings.isSegRepEnabled() && indexSettings.isRemoteStoreEnabled() && indexSettings.isRemoteTranslogStoreEnabled();
+    };
+
```

Comment:
Not a fan of static functions, please avoid as much as possible

Reply:
have kept it in one single place to avoid repeating the condition at multiple places.